### ISSUE TITLE
other subnet mask is invisble

### ISSUE THREAD

**Issue by PhagoZ** [ioBroker/ioBroker.ping#52]
my ioBroker has IP 192.168.0.41 and can´t find my devices which are in area 192.168.31.xxx

Other devices like sonoff via MQTT can communicate between these subnets.
My mask at the primary network is set to 255.255.0.0 which causes the IP for the ioBroker.
In line behind this router, the secound router is with the standard 255.255.255.0 mask because I can´t change it at this device.

I´m using this adapter for my presence control and before my old network was overloaded because of Iot AND normal usage, it worked great. Now I have an IoT Network and a standard.
So our Smartphones are in the standard network 192.168.31.xxx and the ioBroker need to see these devices.

Any idea or workarounds?

Thanks a lot.

**Comment by Apollon77** [comment]
Infact can you ping the wanted IP directly from command line ofn the ioBroker hst? if notz then start checking there first ...  If you can then also the adapter should be able to ...

**Comment by mcm1957** [comment]
I can confirm that theres a problem.
If I configure the adapter with IPs from my subnet - everything is working fine.
If I configure an external node (i.e. 8.8.8.8) the adapter displays false while pinging from the commandline shows normal responses.

With debug enabled the following error ist logged
```
ping.0 | 2022-12-04 20:34:27.700 | debug | STDERR: /bin/ping: 136.243.225.161: Name or service not known
ping.0 | 2022-12-04 20:34:27.679 | debug | System command: /bin/ping -n -w 2 -c 1 136.243.225.161
ping.0 | 2022-12-04 20:34:27.677 | debug | Ping result for 136.243.225.161: false in -ms (Tried 2/3 times)
```

BUT when executing the same command at the commandline it works normally.
Even if I use sudo -u iobroker .... everything works normally.

I'll continue to investigate 

<br class="Apple-interchange-newline">
</body>
</html>

**Comment by mcm1957** [comment]
@PhagoZ 
Could you please check whether a trailing / leading whitespace character at the IP address could be the reason for your problem too ?

**Comment by GermanBluefox** [comment]
Please check with 1.6.1. IPs will be trimmed

**Comment by mcm1957** [comment]
should be fixed with 1.6.2
https://github.com/ioBroker/ioBroker.ping#162-2023-07-19

@Apollon77 
can be closed
